optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/variable/listener/VariableListener.java

```diff
@@ -47,36 +51,36 @@ default boolean requiresUniqueEntityEvents() {
      * @param scoreDirector never null
      * @param entity never null
      */
-    void beforeEntityAdded(ScoreDirector scoreDirector, Entity_ entity);
+    void beforeEntityAdded(ScoreDirector<Object> scoreDirector, Entity_ entity);
```

Comment:
Are we sure this doesn't break the API?

If a user had written:  `method(ScoreDirector<MyRoster> ...)` it wouldn't have compiled
But if a user had written 

```
beforeEntityAdded(ScoreDirector scoreDirector, ...) {
   ((ScoreDirector<MyRoster>) scoreDirector).getMyShiftList()...
   ...
}
```
Would that still compile?

Reply:
Would it compile if the method signature was ScoreDirector<?>

Reply:
I've removed `<Solution_>` from the new `VariableListener`. This should remove all potential for generic corner case problems.